Lib/fontTools/ttLib/tables/_g_l_y_f.py

```diff
@@ -1549,16 +1551,40 @@ def dropImpliedOnCurvePoints(*interpolatable_glyphs: Glyph) -> Set[int]:
     Returns:
         The set of point indices that were dropped if any.
 
+    Raises:
+        ValueError if simple glyphs are not in fact interpolatable because they have
+        different point flags or number of contours.
+
     Reference:
     https://developer.apple.com/fonts/TrueType-Reference-Manual/RM01/Chap1.html
     """
-    assert len(interpolatable_glyphs) > 0
-
+    numContours = None
+    flags = None
     drop = None
-    for glyph in interpolatable_glyphs:
+    simple_glyphs = []
+    for i, glyph in enumerate(interpolatable_glyphs):
+        if glyph.numberOfContours < 1:
+            # ignore composite or empty glyphs
+            continue
+
+        if numContours is None:
+            numContours = glyph.numberOfContours
+        elif glyph.numberOfContours != numContours:
+            raise ValueError(
+                f"Incompatible number of contours for glyph at master index {i}: "
+                f"expected {numContours}, found {glyph.numberOfContours}"
+            )
+
+        if flags is None:
+            flags = glyph.flags
+        elif glyph.flags != flags:
```

Comment:
Should you also check for same `endPtsOfContours`?

Reply:
Right, it I'd conceivable that numContours and flags be equal but endPtsOfContours are different, will update, thanks

Reply:
done https://github.com/fonttools/fonttools/pull/3147/commits/5b93100616199945e23cdcd664242b68ad9bf0f8